### DISCUSSION TITLE
Task Either IO

### DISCUSSION THREAD

**Discussion by FredericLatour** [gcanti/fp-ts#1716] [General]
Hi,
Maybe a stupid question, but why isn't there any TaskIO, TaskIOEither (or IOTask, IOTaskEither) ?
Could just be an alias that would make it possible to clarify from the signature that there is some side effect.
Does it make any sense at all ?

**Comment by steida** [comment]
IO is sync, Task is async

**Comment by samhh** [comment]
`Task` represents an asynchronous side effect, `IO` represents a synchronous side effect. The former essentially subsumes the latter so it would be nearly as redundant as `OptionOption` would be.

If you ever feel compelled to nest an `IO` inside a `Task`, instead lift the `IO` to a `Task` and monadic bind them together using something like [`Task.chainIOK`](https://gcanti.github.io/fp-ts/modules/Task.ts.html#chainiok).

**Comment by FredericLatour** [comment]
So the reasoning is that an asynchronous computation has necessarily side effects ?
Just trying to understand.

**Reply by samhh** [reply]
I can't think of anything pure that should be async. `Lazy` perhaps if it's expensive, but still sync.

**Reply by steida** [reply]
Time means something has an effect that takes time. Time itself is a side effect because it depends on something outside of your code.

**Reply by FredericLatour** [reply]
Makes sense (even though one can make any function async with just a keyword, I can't see indeed what would be the point). 
So, as far as fp-ts is concerned, Task => side effect.
Thanks for clarifying.
